pytoda/smiles/polymer_language.py

```diff
@@ -79,46 +100,118 @@ def update_entity(self, entity: str) -> None:
         Returns:
             None
         """
-        assert (
-            entity.capitalize() in self.entities
-        ), f'Unknown entity was given ({entity})'
-        self.current_entity = entity.capitalize()
+        self.current_entity = self._check_entity(entity)
+        self.reset_initial_transforms()
```

Comment:
why does the entity update require to reset the transforms?

Reply:
I don't think it is!